packages/api/methods.ts

```diff
@@ -205,8 +206,8 @@ export function updateRule(rule) {
   return send('api/rule-update', { rule });
 }
 
-export function deleteRule(id) {
-  return send('api/rule-delete', { id });
+export function deleteRule(rule: Pick<RuleEntity, 'id'>) {
+  return send('api/rule-delete', rule);
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏this is a breaking change. I don't think we should do it as it breaks consistency with the other `delete*` API methods.

Reply:
Correct me if I'm wrong, but I think this is not actually a breaking change. IIRC, if you look at the tests for this file, the test passes in the full rule as "id", so I think the parameter was incorrectly named before. The change from `{ id }` to `rule` is because I did change the type signature of the internal API to match the external one

Reply:
If we want to match other delete APIs, I think that would be a breaking change but maybe a worthwhile one, defer to you

Reply:
Damn.. When reviewing this originally I must have missed it. Yes - we should definitely align this with the other `delete*` methods (i.e. convert it to a `id: string` type). Even if that's a breaking change.